WcaOnRails/app/models/post.rb

```diff
@@ -26,7 +26,7 @@ def edit_path
     if is_crash_course_post?
       Rails.application.routes.url_helpers.delegate_crash_course_edit_path
     else
-      Rails.application.routes.url_helpers.edit_post_path(slug)
+      Rails.application.routes.url_helpers.edit_post_path(id: slug)
```

Comment:
Actually by re-reading the output from byebug it does...
Well idk, it definitely looks like there is a difference in Rails internals between calling this from the model:
```ruby
Rails.application.routes.url_helpers.edit_post_path(slug)
```
which gives this:
```
[22, 31] in /home/fifi/Documents/cube/wca/worldcubeassociation.org/WcaOnRails/app/models/post.rb
   22:     !is_crash_course_post?
   23:   end
   24: 
   25:   def edit_path
   26:     byebug
=> 27:     if is_crash_course_post?
   28:       Rails.application.routes.url_helpers.delegate_crash_course_edit_path
   29:     else
   30:       Rails.application.routes.url_helpers.edit_post_path(id: slug)
   31:     end
(byebug) Rails.application.routes.url_helpers.edit_post_path(slug)
*** ActionController::UrlGenerationError Exception: No route matches {:action=>"edit", :controller=>"posts", :id=>nil, :locale=>"you-can-t-parse-the-system-without-copying-the-1080p-http-circuit"} missing required keys: [:id]

nil
(byebug) Rails.application.routes.default_url_options
{:locale=>:fr}
```

And this from the controller:
```ruby
post_path(@post.slug)
```
which gives this:
```
[27, 36] in /home/fifi/Documents/cube/wca/worldcubeassociation.org/WcaOnRails/app/controllers/posts_controller.rb
   27:     @post.world_readable = true
   28:     @post.author = current_user
   29:     if @post.save
   30:       flash[:success] = "Created new post"
   31:       byebug
=> 32:       redirect_to post_path(@post.slug)
   33:     else
   34:       render 'new'
   35:     end
   36:   end
(byebug) post_path(@post.slug)
"/fr/posts/titi"
(byebug) edit_post_path(@post.slug)
"/fr/posts/titi/edit"
(byebug) Rails.application.routes.default_url_options
{:locale=>:fr}
```